### PR TITLE
Add PlatformServices, initialize it and use in DNX

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/PlatformServicesExtension.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Dnx.Runtime.Common.DependencyInjection
+{
+    internal static class PlatformServicesExtension
+    {
+        public static void AddPlatformServices(this ServiceProvider serviceProvider)
```
Comment (pakrym) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/PlatformServicesExtension.cs]: Add PlatformServices parameter and another overload to call it with PlatformServices.Default?

File: src/Microsoft.Dnx.Runtime.Abstractions/DefaultPlatformServices.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class DefaultPlatformServices : PlatformServices
```
Comment (pakrym) [inline src/Microsoft.Dnx.Runtime.Abstractions/DefaultPlatformServices.cs]: Make internal and expose Create method through PlatformServices?

File: src/Microsoft.Dnx.Host/Bootstrapper.cs [hunk 3]
```diff
@@ -70,6 +70,15 @@ public Task<int> RunAsync(List<string> args, IRuntimeEnvironment env, string app
                 serviceProvider.Add(typeof(IRuntimeEnvironment), env);
 
                 CallContextServiceLocator.Locator.ServiceProvider = serviceProvider;
+
+                PlatformServices.SetDefault(
+                    PlatformServices.Create(
+                        null,
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Host/Bootstrapper.cs]: use optional parameter syntax

**Comment by davidfowl** [comment]
Minor comment then :shipit:

**Comment by per-samuelsson** [comment]
I noticed this was done and merged. So now we have both `PlatformServices` and `CallContextServiceLocator`? Anyone can share a line or two on the semantic differences?

**Comment by davidfowl** [comment]
https://github.com/aspnet/dnx/issues/2927

**Comment by per-samuelsson** [comment]
> #2927

Thanks! Sloppy searching by me obv. Sorry. :bouquet:

**Pull request merged**